File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    
    public int solution(int n, int k) {
        int answer = 0;
        String s = "";
        String sNum = "";
        long nn = n;
        while(n != 0) {
            int num = (int)(n % k);
            n /= k;
            s += String.valueOf(num);
        }
        for(int i = s.length()-1; i >=0; i--) {
            sNum += s.charAt(i);
        }
        String[] str = sNum.split("0");
        for(String ss : str) {
            if(!ss.isEmpty()) {
                long num = Long.parseLong(ss);
                if(isPrime(num)) {
                    answer++;
                }
            }
            
        }
        return answer;
    }
    boolean isPrime(long n) {
        if(n <= 1) return false;
        for(long i = 2; i <=Math.sqrt(n); i++) {
            if(n%i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function solution(n, k) {
    var answer = 0;
    const arr = n.toString(k).split('0').map(Number);
    
    function isPrime(n) {
        if (n < 2) return false;
        const sq = Math.floor(Math.sqrt(n));
        for (let i = 2; i <= sq; i++) {
            if (n % i === 0) return false;
        }
        return true;
    }
    for (const num of arr) {
        if (isPrime(num)) answer++;
    }
    return answer;
}